python/tests/graph/base_test_generators.py

```diff
@@ -1,314 +0,0 @@
-# Copyright 2017 Databricks, Inc.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-from __future__ import absolute_import, division, print_function
-
-from collections import namedtuple
-from contextlib import contextmanager
-from glob import glob
-import itertools
-import os
-import shutil
-import tempfile
-
-import numpy as np
-import tensorflow as tf
-
-from sparkdl.graph.input import *
-import sparkdl.graph.utils as tfx
-
-TestCase = namedtuple('TestCase', ['bool_result', 'err_msg'])
-TestFn = namedtuple('TestFn', ['test_fn', 'description'])
-_GinInfo = namedtuple('_GinInfo', ['gin', 'description'])
-
-
-class TestGenBase(object):
-    def __init__(self, vec_size=17, test_batch_size=231):
-        # Testing data spec
-        self.vec_size = vec_size
-        self.test_batch_size = test_batch_size
-
-        # TensorFlow graph element names
-        self.input_op_name = 'tnsrOpIn'
-        self.feed_names = []
-        self.output_op_name = 'tnsrOpOut'
-        self.fetch_names = []
-
-        # DataFrame column names
-        self.input_col = 'dfInputCol'
-        self.output_col = 'dfOutputCol'
-
-        # Connecting data from Spark to TensorFlow
-        self.input_mapping = {}
-        self.output_mapping = {}
-
-        # When testing against multiple graph inputs,
-        # derive new names for the DataFrame columns and TensorFlow graph elements.
-        self.reset_iomap(replica=1)
-
-        # The basic stage contains the opaque :py:obj:`TFInputGraph` objects
-        # Any derived that override the :py:obj:`build_input_graphs` method will
-        # populate this field.
-        self.input_graphs = []
```

Comment:
(sync'ed offline)
@thunterdb mentioned that the main complaints were that the fields `input_graps` and `test_cases` are modified during test case construction, and that might hurt the code's readability. 
This is, unfortunately, a walk-around due to the limit on Python's unittest framework.